Hazelcast.Net/Hazelcast.Client.Proxy/ClientMapProxy.cs

```diff
@@ -563,8 +631,7 @@ public ISet<TKey> KeySet()
         {
             var request = MapKeySetCodec.EncodeRequest(GetName());
             var result = Invoke(request, m => MapKeySetCodec.DecodeResponse(m).response);
-
-            return ToSet<TKey>(result);
+            return new ReadOnlyLazySet<TKey>(result, GetContext().GetSerializationService());
         }
 
         public IDictionary<TKey, TValue> GetAll(ICollection<TKey> keys)
```

Comment:
Would it be desirable to return lazy dictionary?

Reply:
Yes but lazy dictionary is not so easy to implement. We may improve it on its own PR.